File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int[] result = new int[gain.length + 1]; // Increased the size by 1
        int sum = 0;

        for (int i = 0; i < gain.length; i++) {
            sum += gain[i];
            result[i + 1] = sum; // Store cumulative sum in the next index
        }

        int maxAltitude = 0;
        for (int i = 0;i<result.length;i++) {
            maxAltitude = Math.max(maxAltitude, result[i]);
        }

        return maxAltitude;
    }
}

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
    const arr = [];
    arr.push(0);
    let sum = 0;
    for(let i=0;i<gain.length;i++){
        sum+=gain[i];
        arr.push(sum);
    }
    let max = 0;
    for(let i=0;i<arr.length;i++){
        max = Math.max(arr[i],max);
    }
    return max;
};